onmt/model_builder.py

```diff
@@ -375,9 +383,26 @@ def build_model(model_opt, opt, vocabs, checkpoint):
         # => strict=False when loading state_dict
         strict = not model_opt.update_vocab
 
-        model.load_state_dict(
-            checkpoint, precision=precision, device=device, strict=strict
-        )
+        if "model" in checkpoint.keys():
+            # weights are in the .pt file
+            model.load_state_dict(
+                checkpoint,
+                precision=precision,
+                device=device,
+                strict=strict,
+            )
+        else:
+            # weights are not in the .pt checkpoint but stored in the safetensors file
+            base_name = (
+                opt.train_from[:-3] if opt.train_from[-3:] == ".pt" else opt.train_from
+            )
+            model_path = base_name
```

Comment:
Specific reason for adding this intermediary name?

Reply:
no